__tests__/server.test.js

```diff
@@ -1,37 +1,38 @@
-'use strict';
-
 const supertest = require('supertest');
-const server = require('../server.js');
+const server = require('../src/server.js');
 
 const request = supertest(server.app);
 
-describe('Api Server', () => {
-  it('handles requests to missing routes', async () => {
-    const response = await request.get('/foo');
-    expect(response.status).toEqual(404);
-  });
+describe('Node Server', () => {
+    it('Says Hello World', async () => {
+        // set up the test so it can do a thing
+        // Prepare the server
+        // (see above)
 
-  it('handles errors', async () => {
-    const response = await request.get('/bad');
-    expect(response.status).toEqual(500);
-    expect(response.body.route).toEqual('/bad');
-  });
+        // perform an action, that does the thing
+        // request the / route
 
-  it('handles root path/', async () => {
-    const response = await request.get('/');
-    expect(response.status).toEqual(200);
-    expect(typeof response.body).toEqual('object');
-  });
+    const response = await request.get('/'); // The response is a promise
 
-  it('handles API path /data', async () => {
-    const response = await request.get('/data');
-    expect(response.status).toEqual(200);
-    expect(typeof response.body).toEqual('object');
-  });
+    // assert or expect the result of the action
+    // expect the / route to respond with hello
+    expect(response.status).toBe(200);
+    expect(response.text).toEqual('Hello, World');
+});
 
-  it('stamps routes with middleware', async () => {
+it('Gets Student Data', async () => {
     const response = await request.get('/data');
-    expect(response.status).toEqual(200);
-    expect(response.body.time).toBeDefined();
+
+    expect(response.status).toBe(200);
+    expect(response.body).toEqual({
+      name: 'Tony',
+      role: 'Student',
+    });
   });
+
+  it('should respond 500 on a server error', async () => {
+    const response = await request.get('/throw-error');
```

Comment:
change '/throw-error' to just '/person' because it require an object( {name:name})